app/Services/BaseService.php

```diff
@@ -160,29 +160,30 @@ protected function prompts(): void
     {
         $items = [];
 
-        foreach ($this->defaultPrompts as $prompt) {
-            $this->askQuestion($prompt, $this->useDefaults);
-
-            while ($prompt['shortname'] === 'port' && ! $this->environment->portIsAvailable($this->promptResponses['port'])) {
-                app('console')->error("Port {$this->promptResponses['port']} is already in use. Please select a different port.");
-                $this->askQuestion($prompt);
-            }
-        }
-
-        foreach ($this->prompts as $prompt) {
-            $this->askQuestion($prompt, $this->useDefaults);
-
-            while ($prompt['shortname'] === 'volume' && ! $this->docker->volumeIsAvailable($this->promptResponses['volume'])) {
-                app('console')->error("Volume {$this->promptResponses['volume']} is already in use. Please select a different volume.");
-                $this->askQuestion($prompt);
-            }
-
-            while (Str::contains($prompt['shortname'], 'port') && ! $this->environment->portIsAvailable($this->promptResponses[$prompt['shortname']])) {
-                app('console')->error("Port {$this->promptResponses[$prompt['shortname']]} is already in use. Please select a different port.");
-                $this->askQuestion($prompt);
-            }
-
-            $items[] = $prompt;
+        $questions = array_merge($this->defaultPrompts, $this->prompts);
+
+        foreach ($questions as $prompt) {
+            $items[] = match (true) {
+                Str::contains($prompt['shortname'], 'port') => $this->askQuestion($prompt, $this->useDefaults, validate: function (string $port) {
```

Comment:
Why are we moving to `Str::contains`? It makes it less clear what we're doing.

Reply:
This was kinda duplicated. For the default prompts we were checking `shortname === port` but for the custom prompts (the services might have other port prompts), we were checking `Str::contains(shortname, 'port')`, so I figured it would be better to merge the prompts and just check once with Str::contains, which handles both cases.